[fix] 회원정보 수정기능 관련 수정

Hanjarang opened the pull request:
## 📝 변경 사항
### AS-IS
- 회원 정보 수정 사항의 이미지를 url 로 넘기고 있음
- 회원 정보 수정 사항들을 body 에 JSON raw 형태로 입력받고 있음.

### TO-BE
- 회원정보 수정 시 이미지를 파일 자체로 넘길 수 있도록 수정
- 회원정보 수정 사항을 FormData 형식으로 보낼 수 있도록 수정

## 🔍 테스트
- [ ] 테스트 코드 작성
- [x] API 테스트
- [ ] 로컬 테스트

## 📸 스크린샷
<img width="749" alt="스크린샷 2025-01-21 오전 3 06 08" src="https://github.com/user-attachments/assets/aa28378e-52a6-4233-a3bc-97f39fb87ec5" />
<img width="749" alt="스크린샷 2025-01-21 오전 3 06 17" src="https://github.com/user-attachments/assets/12f59923-abed-4589-bdd4-83c79c0c1436" />


## ✅ 체크리스트
- [x] main/develop 브랜치가 아닌 feature 브랜치에서 작성했나요?
- [ ] 코딩 컨벤션을 준수했나요?
- [ ] 불필요한 코드는 제거했나요?
- [ ] 주석은 충분히 작성했나요?
- [ ] 테스트는 완료했나요?

## 🙋‍♂️ 리뷰어에게
확인부탁드립니다.

Hanjarang committed:
[fix] 회원정보 수정기능 관련 수정

- 회원정보 수정 시 이미지를 파일 자체로 넘길 수 있도록 수정

- 수정 사항을 FormData 형식으로 보낼 수 있도록 수정

Seokjin-K commented on src/main/java/com/momo/user/service/UserService.java:
```diff
@@ -310,8 +313,9 @@ public void updateUser(UserUpdateRequest updateRequest) {
     userRepository.save(user);
```
트랜잭션 거신 상태로 엔티티를 조회하셨으면 해당 엔티티는 이미 영속성 컨텍스트에 올려져 있는 상태이기 때문에 따로 save가 필요 없을 거예요. 해당 트랜잭션이 종료될 때 엔티티에 변경사항이 있다면 자동으로 커밋해줍니다.

Seokjin-K commented on src/main/java/com/momo/user/service/UserService.java:
```diff
@@ -310,8 +313,9 @@ public void updateUser(UserUpdateRequest updateRequest) {
     // Profile 엔티티 업데이트
-    if (updateRequest.getProfileImageUrl() != null) {
-      profile.setProfileImageUrl(updateRequest.getProfileImageUrl());
+    if (profileImage != null && !profileImage.isEmpty()) {
```
기존 프로필 이미지는 삭제가 필요해 보입니다.

Seokjin-K reviewed (commented):
수고하셨습니다.

Hanjarang committed:
[fix] 회원정보 이미지 수정시 기존 이미지 삭제

- 회원정보 이미지 수정시 기존 이미지를 삭제할 수 있도록하는 로직을 추가

- updateUser 에 @Transactional가 적용된 상태로 불필요한 save() 로직 제거

Hanjarang committed:
[fix] UserService 불필요한 로직 제거

- 더 이상 사용되지않는 로직 제거

Hanjarang committed:
Merge branch 'develop' into fix/회원정보수정_관련
